core/templates/dev/head/pages/exploration_editor/simple_editor_tab/QuestionIdService.js

```diff
@@ -0,0 +1,40 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that contains common functionalities for creating
+ * unique identifiers for sidebar items and editor field headers.
+ */
+
+oppia.factory('QuestionIdService', [function() {
+  // Strip the space, and set to lower case.
+  var strip = function(string) {
```

Comment:
this is used to create the rest of the elements ids as a base, like this:
   - question.getId()-subfieldlabel.tolowercase and no space. -> this is a h3 field header id. 
   - sidebaritem-question.getId()-subfieldId -> this a sidebar item id

Now because sometimes i need to get the questionId from subfield id i use the hyphen as a separator, so if the questionId have a hyphen in it, it would break mainly the 'getParentQuestionId' method.